pkg/io/storage/postgre_test.go

```diff
@@ -0,0 +1,178 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/golang-migrate/migrate/v4"
+
+	_ "github.com/golang-migrate/migrate/v4/database/postgres"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	postgreURL = "postgres://postgres:postgres@localhost:5432/postgres?sslmode=disable"
+	migration  = "file://../../../scripts/migrations"
+)
+
+func TestPostgreVault_SaveData(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		db, err := sqlx.ConnectContext(ctx, "postgres", postgreURL)
+		require.NoError(t, err)
+		defer disconnectPDB(db)
+
+		migrateUp()
+		defer migrateDown()
+
+		err = db.Ping()
+		require.NoError(t, err)
+
+		d := NewPostgreVault(db)
+		err = d.SaveData([]byte("k1234"), []byte("value1234"))
+		require.NoError(t, err)
+	})
+	t.Run("success update if try set repeated key in db", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		db, err := sqlx.ConnectContext(ctx, "postgres", postgreURL)
+		require.NoError(t, err)
+		defer disconnectPDB(db)
+
+		migrateUp()
+		defer migrateDown()
+
+		err = db.Ping()
+		require.NoError(t, err)
+
+		d := NewPostgreVault(db)
+		err = d.SaveData([]byte("k1234"), []byte("value1234"))
+		require.NoError(t, err)
+
+		err = d.SaveData([]byte("k1234"), []byte("value"))
+		require.NoError(t, err)
+
+		data, err := d.ReadData([]byte("k1234"))
+		require.NoError(t, err)
+		require.EqualValues(t, "value", string(data))
+	})
+	t.Run("key nil error if try set nil value into db", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		db, err := sqlx.ConnectContext(ctx, "postgres", postgreURL)
+		require.NoError(t, err)
+		defer disconnectPDB(db)
+
+		migrateUp()
+		defer migrateDown()
```

Comment:
It would be more expressive if at first:
- do migration stuff
- create a db connection and do your work.
```suggestion
		migrateUp()
		defer migrateDown()
		
		db, err := sqlx.ConnectContext(ctx, "postgres", postgreURL)
		require.NoError(t, err)
		defer disconnectPDB(db)
```